File: forest/the-agrippa-defense.lua
loop
    local enemy = self:findNearestEnemy()
    if enemy then
        d = self:distanceTo(enemy)
        if d < 5 then
            if self:isReady("cleave") then
                self:cleave(enemy)
            else
                self:attack(enemy)
            end
        else
            self:shield()
        end
    end
end

File: forest/the-agrippa-defense.clj
(while true
  (def e (.findNearestEnemy this))
  (if e
    (when true
      (def d (.distanceTo this e))
      (if (< d 5)
        (if (.isReady this "cleave")
          (.cleave this e)
          (.attack this e))
        (.wait this 0.25)))
    (.wait this 0.25)))
